src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -963,11 +963,17 @@ public static ulong ToUInt64(string s)
 
         public static float ToSingle(string s)
         {
-            s = TrimString(s);
-            if (s == "-INF") return float.NegativeInfinity;
-            if (s == "INF") return float.PositiveInfinity;
-            float f = float.Parse(s, NumberStyles.AllowLeadingSign | NumberStyles.AllowDecimalPoint | NumberStyles.AllowExponent, NumberFormatInfo.InvariantInfo);
-            if (f == 0 && s[0] == '-')
+            ReadOnlySpan<char> roS = s.AsSpan().Trim(WhitespaceChars);
```

Comment:
Capitalization looks kind of weird here:

```suggestion
            ReadOnlySpan<char> ros = s.AsSpan().Trim(WhitespaceChars);
```

Reply:
I'd say name is weird, name should reflect content not the type

Reply:
Is it better @krwq ?
``` cs
public static float ToSingle(string s)
        {
            ReadOnlySpan<char> value = s.AsSpan().Trim(WhitespaceChars);
            switch (value)
            {
                case "-INF":
                    return float.NegativeInfinity;
                case "INF":
                    return float.PositiveInfinity;
                default:
                    float f = float.Parse(value, NumberStyles.AllowLeadingSign | NumberStyles.AllowDecimalPoint | NumberStyles.AllowExponent, NumberFormatInfo.InvariantInfo);
                    if (f == 0 && value[0] == '-')
                    {
                        return -0f;
                    }
                    return f;
            }
        }
```